### PR TITLE
[BUG] mongoDB Session schema expiry index not working

### LINKED ISSUE #296: [BUG] Sessions are removed from mongoDB far earlier than expiry
## Bug Report Body:
Currently, new sessions created for each game are being removed before the game ends. The sessions are expected to be removed on expiry after 7 days.
By removing the sessions from the database faster than the game itself, it causing the error of where the code is trying to find a non-existent session to work with. This was because the sessionID was no longer associated with a session (as it is removed)

Process to find the issue:

We have located the source of the issue to:
createdAt: {type: Date, expires: config, default: Date.now},

One small note is that `expires: config` should be changed to `expires: config.session_expiry`. However this does not solve the issue. 
Firstly, the `expires: config.session_expiry` part was entirely removed. However, it still deletes the sessions from the database.

Then the entire line was removed `createdAt: {type: Date, expires: config, default: Date.now}`,
This stopped removing the sessions from the database and the null session error stop occurring.

### To Reproduce the bug:
Note: this bug does not always present itself visibly. And so the mongoDB sessions log must be looked at.

1. Run the Yumble game, until the result screen.
2. Keep track of the session created on the mongoDB sessions log.
3. The sessions should at some point be removed.
4. Click on the see results button.
5. The pie chart will be missing.
6. Expected behaviour: The pie chart should not be missing, and the session should not be removed from MongoDB.

Screenshots - 
Error of the session that could not be found being used.
image

MongoDB missing the session.
image

* OS: Windows
* Browser: chrome and Edge

### PR DESCRIPTION
**Pull request by parfei**
### The problem
**Mongodb does not allow you to modify indexes, especially the TTL value.** We had assumed that you could, so everyime we saved a `Session` document with a certain expiry value, we assumed that value would override the old one. It did not.

We had first tested the creation of an index with 30 seconds. Therefore, the index TTL value was always being referenced to as 30 seconds - documents were expiring at least 30 seconds and caused `SocketSession` null pointer exceptions as a result.

### Solution
* Drop and recreate the index everytime you restart the app. This will allow the user to define the `expireAtSeconds` value to whatever they want, which makes things easier. HOWEVER, **please do not test with `myFirstDatabase`, as everytime you drop and create an index for MongoDB, it blocks all over operations on the database**. So you can use `testDatabase` if you want
* Extracted expiry time to `.env`. Added the example to `.env.example`. No longer need config file (no idea why I did this in the first place tbh)
* Created tests for index creation, dropping and `Session` expiry
* Refactored mongo methods out of `index.js`

As a result, `myFirstDatabase` expiry for `Session`'s `createdAt` index is 1 week. It has already been updated anyway.
image

I have tested _both_ when you first create an index, and when you try to refresh the index value when it already exists.

Also, I wrote some notes here: https://github.com/we-are-number-1/yumble/wiki/Yumble-Architecture-&-Design#database---mongodb and https://github.com/we-are-number-1/yumble/wiki/MongoDB-Atlas

Closes #296

### CODE REVIEW

**Comment by parfei** [comment]
For more info, check out https://docs.mongodb.com/manual/core/index-ttl/
https://stackoverflow.com/questions/29525010/mongoose-expires-property-not-working-properly

File: backend/src/index.js [hunk 1]
```diff
@@ -24,18 +23,9 @@ app.use(express.json());
...
-    {useNewUrlParser: true, useUnifiedTopology: true},
-    (err) => {
-      if (err) {
-        throw err;
-      } else {
-        console.log(`Successfully connected to MongoDB Atlas.`);
-      }
-    },
-);
+// MongoDB Setup
```
Comment (parfei) [inline backend/src/index.js:26]: Refactored into `backend/src/mongo/index.js`

**Review by UltimaKuma — APPROVED** [review]
LGTM though one of the tests are failing due to timeout (src/sockets/__tests__/joinRoom.test.js). I dont think its due to this PR and all the tests work in the other testing PR so it should be fine.

**Review by jteo97 — APPROVED** [review]
Great work. I have tested it myself and it works, it deletes the document on the database after the specified seconds in the env file. Make sure to notify the Heroku admins to add a config vars on the deployment server called SESSION_EXPIRY.

**Pull request merged**